ui/tree/css/_tree.scss

```diff
@@ -319,43 +319,68 @@
   line {
     display: block;
     padding-inline-start: 7px;
+    position: relative;
   }
 
   &-column line {
     margin: 2px 0;
   }
 
-  lines lines move {
-    font-size: 13px;
+  line > branch {
+    border-inline-start: 2px solid $c-border;
+    margin-inline-start: -9px;
+    position: absolute;
+    width: 8px;
+    height: 100%;
   }
 
-  lines lines {
-    margin-inline-start: 1px;
+  line:last-child > branch,
+  line.expand > branch {
+    height: calc(0.8em + 2px);
   }
 
-  lines lines::before {
+  line > branch::before {
+    margin-top: 0.8em;
     content: ' ';
     border-top: 2px solid $c-border;
     position: absolute;
-    margin-inline-start: -11px;
-    width: 9px;
+    width: 7px;
     height: 6px;
   }
 
-  lines line::before {
-    margin-top: 0.65em;
-    margin-inline-start: -8px;
-    content: ' ';
-    border-top: 2px solid $c-border;
-    position: absolute;
-    width: 8px;
-    height: 6px;
+  &-column line > branch {
+    margin-top: -2px;
+    height: calc(100% + 2px);
+  }
+
+  lines lines move {
+    font-size: 13px;
+  }
+
+  lines lines {
+    margin-inline-start: 1px;
   }
 
   lines lines:last-child {
     margin-bottom: 0;
   }
 
+  lines.collapsed > * {
+    display: none;
```

Comment:
instead of hiding them, can we avoid rendering them altogether? And maybe get some perf gains in deep trees.

Reply:
Done.